packages/ebayui-core-react/src/ebay-lightbox-dialog/__tests__/index.stories.tsx

```diff
@@ -357,7 +352,36 @@ export const Expressive: StoryFn<typeof EbayLightboxDialog> = (args) => {
                 </p>
             </EbayLightboxDialog>
         </div>
-    )
-}
+    );
+};
+
+export const DebugOnClose: StoryFn<typeof EbayLightboxDialog> = (args) => {
+    const [open, setOpen] = useState(true);
+    const [showButton, setShowButton] = useState(true);
+
+    return (
+        <div>
+            <button className="btn btn--secondary" onClick={() => setOpen(!open)}>
+                Open Dialog
+            </button>
+            <EbayLightboxDialog
+                {...args}
+                bannerImgSrc="http://ir.ebaystatic.com/cr/v/c1/skin/image-treatment/mountain.jpeg"
+                bannerImgPosition="top"
+                open={open}
+                onOpen={() => action("onOpen")()}
+                onClose={() => {
```

Comment:
This onClose function is causing not being able to simulate the issue. We need to wrap it in a `useCallback()` so it will not change the reference on every render and the `useEffect()` will not trigger the removeEventListener